test/e2e/cli/net_rm_test.go

```diff
@@ -0,0 +1,151 @@
+// You may not use this file except in compliance with the License.
+
+package cli_test
+
+import (
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	fcmd "kraftkit.sh/test/e2e/framework/cmd"
+	fcfg "kraftkit.sh/test/e2e/framework/config"
+)
+
+var _ = Describe("kraft net rm", func() {
+	var cmd *fcmd.Cmd
+
+	var stdout *fcmd.IOStream
+	var stderr *fcmd.IOStream
+
+	var cfg *fcfg.Config
+
+	BeforeEach(func() {
+		stdout = fcmd.NewIOStream()
+		stderr = fcmd.NewIOStream()
+
+		cfg = fcfg.NewTempConfig()
+
+		cmd = fcmd.NewKraftPrivileged(stdout, stderr, cfg.Path())
+		cmd.Args = append(cmd.Args, "net", "rm", "--log-level", "info", "--log-type", "json")
+	})
+
+	When("invoked without flags or positional arguments", func() {
+		It("should print an error and exit with an error", func() {
+			err := cmd.Run()
+			Expect(err).To(HaveOccurred())
+
+			Expect(stderr.String()).To(BeEmpty())
+			Expect(stdout.String()).To(MatchRegexp(`^{"level":"error","msg":"accepts 1 arg\(s\), received 0"}\n`))
+		})
+	})
+
+	When("invoked with two positional arguments", func() {
+		BeforeEach(func() {
+			cmd.Args = append(cmd.Args, "some-arg")
+			cmd.Args = append(cmd.Args, "some-other-arg")
+		})
+
+		It("should print an error and exit with an error", func() {
+			err := cmd.Run()
+			Expect(err).To(HaveOccurred())
+
+			Expect(stderr.String()).To(BeEmpty())
+			Expect(stdout.String()).To(MatchRegexp(`^{"level":"error","msg":"accepts 1 arg\(s\), received 2"}\n$`))
+		})
+	})
+
+	When("invoked with the --help flag", func() {
+		BeforeEach(func() {
+			cmd.Args = append(cmd.Args, "--help")
+		})
+
+		It("should print the command's help", func() {
+			err := cmd.Run()
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(stderr.String()).To(BeEmpty())
+			Expect(stdout.String()).To(MatchRegexp(`^Remove a network\n`))
+		})
+	})
+
+	// Requires root privileges
+	When("invoked with the --help flag", func() {
+		BeforeEach(func() {
+			cmd.Args = append(cmd.Args, "--help")
+		})
+
+		It("should print the command's help", func() {
+			err := cmd.Run()
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(stderr.String()).To(BeEmpty())
+			Expect(stdout.String()).To(MatchRegexp(`^Remove a network\n`))
+		})
+	})
+
+	// Requires root privileges
+	When("invoked with one positional argument, valid driver, and a valid network", func() {
+		BeforeEach(func() {
+			// Create the network
+			stdout = fcmd.NewIOStream()
+			stderr = fcmd.NewIOStream()
+			cmd = fcmd.NewKraftPrivileged(stdout, stderr, cfg.Path())
+			cmd.Args = append(cmd.Args, "net", "create", "--log-level", "info", "--log-type", "json")
+			cmd.Args = append(cmd.Args, "--driver", "bridge")
+			cmd.Args = append(cmd.Args, "--network", "172.46.0.1/24")
+			cmd.Args = append(cmd.Args, "test-rm-0")
+
+			err := cmd.Run()
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stderr.String()).To(BeEmpty())
+
+			// Reset args
```

Comment:
(Applies all tests suites)
This same command is set in the global `BeforeEach`, so instead of overriding it, then reset it, couldn't we simply use another var for the other setup commands and avoid those reset steps?

Reply:
Makes sense, will do for all